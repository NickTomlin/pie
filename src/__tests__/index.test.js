import React from 'react'
import {Pie, Slice} from '../index'
import renderer from 'react-test-renderer'

test('Renders an svg pie chart', () => {
  const component = renderer.create(
    <Pie>
      <Slice percent={0.1} fill='Coral' />
      <Slice percent={0.7} fill='CornFlowerBlue' />
      <Slice percent={0.2} fill='#00ab6b' />
    </Pie>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Supports donut option', () => {
  const component = renderer.create(
    <Pie donut>
      <Slice percent={0.1} fill='Coral' />
      <Slice percent={0.7} fill='CornFlowerBlue' />
      <Slice percent={0.2} fill='#00ab6b' />
    </Pie>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
