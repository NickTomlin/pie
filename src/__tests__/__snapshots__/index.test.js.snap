// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an svg pie chart 1`] = `
<svg
  style={
    Object {
      "overflow": "visible",
      "transform": "rotate(-90deg)",
    }
  }
  viewBox="-1 -1 2 2"
>
  <path
    d="M 1 0 A 1 1 0 0 1 0.8090169943749475 0.5877852522924731 L 0 0"
    fill="Coral"
  />
  <path
    d="M 0.8090169943749475 0.5877852522924731 A 1 1 0 1 1 0.30901699437494723 -0.9510565162951536 L 0 0"
    fill="CornFlowerBlue"
  />
  <path
    d="M 0.30901699437494723 -0.9510565162951536 A 1 1 0 0 1 1 -2.4492935982947064e-16 L 0 0"
    fill="#00ab6b"
  />
</svg>
`;
